### ISSUE TITLE
`pyreverse` fails with `ModuleNotFoundError`

### ISSUE THREAD

**Issue by crazyscientist** [PyCQA/pylint#4186]
### Steps to reproduce

Given a directory structure like this (assuming CWD is `/tmp`):
```
$ tree foo/
foo/
├── __init__.py
└── processors
    ├── base.py
    ├── buildservice
    │   └── __init__.py
    └── __init__.py

2 directories, 4 files
```
In my actual project these files contain code, but for the purpose of reproduction is is sufficient to have all empty files except for `processors/buildservice/__init__.py`. A single line this this is sufficient:
```python
import re
```

### Current behavior
Pyreverse 2.7.x seems to have problems with handling of the module structure. The traceback is identical for `x in (0, 2)`:
```
$ pyreverse foo/processors
parsing foo/processors/__init__.py...
parsing /tmp/foo/processors/base.py...
parsing /tmp/foo/processors/__init__.py...
parsing /tmp/foo/processors/buildservice/__init__.py...
Traceback (most recent call last):
  File "/home/andi/virtualenvs/smelt/bin/pyreverse", line 8, in <module>
    sys.exit(run_pyreverse())
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/__init__.py", line 37, in run_pyreverse
    PyreverseRun(sys.argv[1:])
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/main.py", line 188, in __init__
    sys.exit(self.run(args))
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/main.py", line 206, in run
    diadefs = handler.get_diadefs(project, linker)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/diadefslib.py", line 236, in get_diadefs
    diagrams = DefaultDiadefGenerator(linker, self).visit(project)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/utils.py", line 210, in visit
    self.visit(local_node)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/utils.py", line 207, in visit
    methods[0](node)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/diadefslib.py", line 162, in visit_module
    self.linker.visit(node)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/utils.py", line 210, in visit
    self.visit(local_node)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/utils.py", line 207, in visit
    methods[0](node)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/pylint/pyreverse/inspector.py", line 247, in visit_import
    relative = modutils.is_relative(name[0], context_file)
  File "/home/andi/virtualenvs/smelt/lib/python3.8/site-packages/astroid/modutils.py", line 581, in is_relative
    parent_spec = importlib.util.find_spec(name, from_file)
  File "/usr/lib/python3.8/importlib/util.py", line 94, in find_spec
    parent = __import__(parent_name, fromlist=['__path__'])
ModuleNotFoundError: No module named 'processors'
```
pyreverse behaves the same when calling `pyreverse foo`.

However this seems to work:

```
$ cd foo/
$ pyreverse processors
parsing processors/__init__.py...
parsing /tmp/foo/processors/base.py...
parsing /tmp/foo/processors/__init__.py...
parsing /tmp/foo/processors/buildservice/__init__.py...
```
But I'm not too eager to call pyreverse on every sub-module, I'd actually prefer to have it run once for my project.

### Expected behavior
Pyreverse 2.6.x handles my actual project and this example correctly howevr. Pyreverse 2.7 should draws the diagram correctly, too.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.0
astroid 2.5
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0]

```
```
pylint 2.7.2
astroid 2.5.1
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0]
```

**Comment by DudeNr33** [comment]
This seems to be a ``astroid`` bug instead of a ``pylint``/``pyreverse`` bug.

A little research:
The bug occurs from Pylint V2.7.0 onwards.
The predecessor Pylint V2.6.2 works fine.

Pylint V2.7.0 requires ``astroid`` V2.5, while Pylint V2.6.2 uses ``astroid`` V2.4.2.
By forcing both libraries to specific versions, the result is:
Pylint V2.7.0 + astroid V2.4.2 -> no error
Pylint V2.6.2 + astroid V2.5 -> error

So the problem must be somewhere in the change ``@@ -614,16 +572,22 @@ def is_relative(modname, from_file)`` of ``astroid.modutils``.

https://github.com/PyCQA/astroid/compare/astroid-2.4.2%E2%80%A6astroid-2.5

**Comment by DudeNr33** [comment]
Ok so basically the ``astroid.modutils.is_relative`` method does not work for any import in a package more than two levels deep.
A [corresponding astroid issue](https://github.com/PyCQA/astroid/issues/930) already exists.

**Comment by Pierre-Sassoulas** [comment]
Apparently it's due to https://github.com/PyCQA/astroid/pull/857/files. The analysis by doublethefish is very thorough.

**Comment by swight-prc** [comment]
This is still an issue, because pylint still requires the version of astroid that predates the fix. I've installed the new version anyway, and nothing has broken so far, but I wanted to bring it up just the same.

**Comment by Pierre-Sassoulas** [comment]
You were right to manually set astroid this is a nice temporary fix. You can even use ``2.6.0-dev0``. We're going to release astroid 2.6.0 and pylint 2.9.0 (I hope this week-end), that will be the real fix.

**Comment by thedewpoint** [comment]
> This is still an issue, because pylint still requires the version of astroid that predates the fix. I've installed the new version anyway, and nothing has broken so far, but I wanted to bring it up just the same.

@swight-prc how did you override this to get it working? I'm trying to do the same

**Comment by Pierre-Sassoulas** [comment]
You can use pylint 2.9.3 now. Or you Can force the version off astroid to 2.6.0+ but other problems might appear.

**Comment by swight-prc** [comment]
> > This is still an issue, because pylint still requires the version of astroid that predates the fix. I've installed the new version anyway, and nothing has broken so far, but I wanted to bring it up just the same.
> 
> @swight-prc how did you override this to get it working? I'm trying to do the same

Sorry I missed this; To override it, I just installed the newer version with pip. That probably wouldn't work in a pipeline, but my current project doesn't have any CD going on, so that wasn't an issue for me yet.
